lib/utils/apiUtils.js

```diff
@@ -26,15 +26,15 @@ function checkRequiredEnvVariables() {
 }
 
 // Get Tokens for the first time with an authorization code
-async function getAccessToken(axiosInstance, firmId, authCode) {
+async function getAccessToken(firmId, authCode) {
   try {
     const redirectUri = "urn%3Aietf%3Awg%3Aoauth%3A2.0%3Aoob";
     const grantType = "authorization_code";
     let requestDetails = {
       method: "POST",
       url: `${BASE_URL}/f/${firmId}/oauth/token?client_id=${process.env.SF_API_CLIENT_ID}&client_secret=${process.env.SF_API_SECRET}&redirect_uri=${redirectUri}&grant_type=${grantType}&code=${authCode}`,
     };
-    const response = await axiosInstance(requestDetails);
+    const response = await axios(requestDetails);
```

Comment:
@delphine-demeulenaere This line was the issue.

When I prepared the staging update, I replaced this line as well to use the new `axiosInstance`, but it wasn't using it's equivalent before.

So I "revert" it to use it `axios` module directly, without the instances we create for other API calls. That is, with no interceptors in place to check existing tokens and refresh tokens if needed.

All other changes are unrelated, to improve redability, isolate logic into it's own class and write test for the authorization process :slightly_smiling_face:

Reply:
Oh, I see! That makes sense. I don’t think I would have discovered this issue on my own though 😅 I tested it and it seems to be working as expected! Thank you Rufo for addressing this so quickly